### ISSUE TITLE
Operator bundle resource descriptions should come from golang doc comments

### ISSUE THREAD

**Issue by babbageclunk** [Azure/azure-service-operator#1416]
**Describe the current behavior**
At the moment the descriptions of the resources are in the template ClusterServiceVersion document `config/operator-bundle/bases/azure-service-operator.clusterserviceversion.yaml` (under `.spec.customresourcedefinitions.owned`). That means that if we add a resource but forget to add it to the yaml, it won't have any description in the Openshift resource UI.

**Describe the improvement**
We should populate that section of the CSV from the doc comments on the Go resource types.  
* The existing doc comments aren't great, they should be replaced with the ones from the CSV.
* Add another build target depending on `manifests` that collects those up and writes them into a generated CSV yaml file in `config/operator-bundle-populated/bases`. (`controller-gen` puts the doc comments into each CRD yaml at `.spec.validation.openAPIV3Schema.description` .)
* Change `generate-operator-bundle` to use that generated yaml as the base rather than the existing template one.

**Comment by babbageclunk** [comment]
From discussion on #1410

**Comment by matthchr** [comment]
The operatorhub CSV (Cluster Service Version) is a document that describes what the operatorhub bundle output looks like. Right now this file is handcrafted for ASO v1. For ASOv2, we would like to autogenerate this file so that we can pull the Resource descriptions from the Go types. This would probably be similar to the `gen-kustomize` target - possibly a new mode called `gen-operatorhubmanifest` or something.

**Comment by sakthi-vetrivel** [comment]
@babbageclunk should we push to next milestone?

**Comment by theunrepentantgeek** [comment]
Confirmed with offline discussion: this is still needed and should be done at the same time as #1967.
Moving it to the same milestone.

**Comment by matthchr** [comment]
Status is still the same as above: Waiting on #1967

**Comment by matthchr** [comment]
Status is still the same as above: Waiting on https://github.com/Azure/azure-service-operator/issues/1967

**Comment by matthchr** [comment]
Still waiting on #1967

**Comment by matthchr** [comment]
Still waiting on https://github.com/Azure/azure-service-operator/issues/1967

**Comment by matthchr** [comment]
Closing in favor of tracking operator bundle stuff via #1967
